src/Filters/coefficients.jl

```diff
@@ -45,11 +47,9 @@ Base.promote_rule(::Type{ZeroPoleGain{D,Z1,P1,K1}}, ::Type{ZeroPoleGain{D,Z2,P2,
 Base.inv(f::ZeroPoleGain{D}) where {D} = ZeroPoleGain{D}(f.p, f.z, inv(f.k))
 
 function Base.:^(f::ZeroPoleGain{D}, e::Integer) where {D}
-    if e < 0
-        return inv(f^-e)
-    else
-        return ZeroPoleGain{D}(repeat(f.z, e), repeat(f.p, e), f.k^e)
-    end
+    ae = uabs(e)
+    res = ZeroPoleGain{D}(repeat(f.z, ae), repeat(f.p, ae), f.k^ae)
+    return e < 0 ? inv(res) : res
```

Comment:
Would this fix the overflow in computing `k` in the test?
```suggestion
    if e < 0
        return ZeroPoleGain{D}(repeat(f.p, ae), repeat(f.z, ae), inv(f.k)^ae)
    else
        return ZeroPoleGain{D}(repeat(f.z, ae), repeat(f.p, ae), f.k^ae)
    end
```